# This config file is used by ros2_control
controller_manager:
  ros__parameters:
    update_rate: 100  # Hz

    panda_arm_controller:
      type: joint_trajectory_controller/JointTrajectoryController


    panda_hand_controller:
      type: position_controllers/GripperActionController


    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

panda_arm_controller:
  ros__parameters:
    joints:
      - panda_joint1
      - panda_joint2
      - panda_joint3
      - panda_joint4
      - panda_joint5
      - panda_joint6
      - panda_joint7
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    allow_nonzero_velocity_at_trajectory_end: true
    open_loop_control: false
    state_publish_rate: 50.0
    action_monitor_rate: 20.0
    gains:
      panda_joint1: {p: 100.0, i: 0.01, d: 1.0}
      panda_joint2: {p: 100.0, i: 0.01, d: 1.0}
      panda_joint3: {p: 100.0, i: 0.01, d: 1.0}
      panda_joint4: {p: 100.0, i: 0.01, d: 1.0}
      panda_joint5: {p: 100.0, i: 0.01, d: 1.0}
      panda_joint6: {p: 100.0, i: 0.01, d: 1.0}
      panda_joint7: {p: 100.0, i: 0.01, d: 1.0}
    interpolation_method: splines
    constraints:
      goal_time: 60.0
      stopped_velocity_tolerance: 2.0
      trajectory:
        path_tolerance: 5.0
        goal_tolerance: 1.0
        min_goal_time: 1.0
panda_hand_controller:
  ros__parameters:
    joint: panda_finger_joint1
